x-pack/plugins/fleet/server/services/output.ts

```diff
@@ -202,23 +233,24 @@ class OutputService {
       data.output_id = options?.id;
     }
 
-    const newSo = await soClient.create<OutputSOAttributes>(SAVED_OBJECT_TYPE, data, {
+    if (output.ssl) {
+      data.ssl = JSON.stringify(output.ssl);
+    }
+
+    const newSo = await this.encryptedSoClient.create<OutputSOAttributes>(SAVED_OBJECT_TYPE, data, {
       overwrite: options?.overwrite || options?.fromPreconfiguration,
       id: options?.id ? outputIdToUuid(options.id) : undefined,
     });
 
-    return {
-      id: options?.id ?? newSo.id,
-      ...newSo.attributes,
-    };
+    return outputSavedObjectToOutput(newSo);
```

Comment:
So is `newSo.ssl` unencrypted here? (but has been encrypted when written to ES?)

Reply:
Yes that is correct, when you use `dangerouslyExposeValue` in the saved object declaration the wrapped saved object client return the decrypted values